core/slider.js

```diff
@@ -0,0 +1,147 @@
+        if (this && this.slider) {
+            this.slider.mouseDown_ = true;
+            if (this.slider.onClickEvent) {
+                this.slider.onClickEvent();
+            }
+            if (this.moveToPoint) {
+                this.updatePosition(ev.clientX);
+            }
+        }
+    })
+
+    Blockly.bindEvent_(window, "mouseup", this, function (ev) {
+        if (this && this.slider && this.slider.mouseDown_) {
+            this.slider.mouseDown_ = false;
+        }
+    })
+
+    Blockly.bindEvent_(window, "mousemove", this, function(ev) {
+        if (this && this.slider && this.slider.mouseDown_) {
+            this.updatePosition(ev.clientX);
+        }
+    })
+}
+
+Blockly.Slider.prototype.updatePosition = function(clientX) {
+    const rect = (this.slider.element_).getBoundingClientRect();
+    let value = clientX - rect.left;
+    const sliderWidth = (this.slider.element_).clientWidth;
+    const pxPerUnit = (this.slider.max_ - this.slider.min_) / sliderWidth;
+    value = value * pxPerUnit;
+    if (this.slider.step_ > 1) {
+        value = Math.round((value - this.slider.min_) / this.slider.step_) * this.slider.step_ + this.slider.min_;
+    } else { // If a step is less than one, multiplying it by the number of notches may result in floating point math errors
+        value = Math.round(((value - this.slider.min_) / this.slider.step_) / (Math.round(1000 / this.slider.step_) / 1000)) + this.slider.min_;
+    }
+    value = Math.min(Math.max(value, this.slider.min_), this.slider.max_);
+    this.slider.setValue(value);
+    if (this.slider.onChangeEvent) {
+        this.slider.onChangeEvent(value);
+    }
+}
+
+Blockly.Slider.prototype.setVisible = function (visible) {
+    this.getElement().style.display = visible ? '' : 'none';
+};
+
+Blockly.Slider.prototype.setMinimum = function (min) {
+    this.min_ = min;
+};
+
+Blockly.Slider.prototype.setMaximum = function (max) {
+    this.max_ = max;
+};
+
+Blockly.Slider.prototype.setStep = function (step) {
+    this.step_ = step;
+};
+
+Blockly.Slider.prototype.setMoveToPointEnabled = function (enabled) {
+    this.moveToPoint = enabled;
+}
+
+Blockly.Slider.prototype.setValue = function (value) {
+    this.value_ = value;
+    value = Math.min(Math.max(value, this.min_), this.max_)
+    var sliderWidth = this.element_.clientWidth;
+    var pxPerUnit = sliderWidth / (this.max_ - this.min_);
+    value = (value - this.min_) * pxPerUnit - (this.thumb_.clientWidth / 2);
+    this.thumb_.style.left = value + "px";
+};
+
+Blockly.Slider.prototype.getValue = function () {
+    return this.value_;
+};
+
+Blockly.Slider.prototype.render = function (parentElement, opt_beforeNode) {
+    if (!this.element_) {
+        this.createDom();
+    }
+    parentElement.insertBefore(this.element_, opt_beforeNode || null);
+};
+
+Blockly.Slider.prototype.createDom = function () {
+    this.element_ = document.createElement('div');
+    this.element_.classList.add('slider');
```

Comment:
Can't use classList, IE doesn't support it. I think Blockly has a helper function for this